sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -448,6 +452,33 @@ object GpuFilter {
     }
   }
 
+  def filterAndCloseWithRetry(input: SpillableColumnarBatch,
+      boundCondition: GpuTieredProject,
+      numOutputRows: GpuMetric,
+      numOutputBatches: GpuMetric,
+      opTime: GpuMetric): Iterator[ColumnarBatch] = {
+    val ret = withRetry(input, splitSpillableInHalfByRows) { sb =>
```

Comment:
Do we need to avoid retries if there are non-deterministic expressions in boundCondition?

Reply:
You are right. Oops...